lib/post_creator.rb

```diff
@@ -381,6 +381,11 @@ def auto_close
           locale: SiteSetting.default_locale
         )
       )
+
+      if SiteSetting.auto_close_topics_create_linked_topic?
+        # enqueue a job to create a linked topic
+        Jobs.enqueue_in(5.seconds, :create_linked_topic, post_id: @post.id)
```

Comment:
Question: Why in 5 seconds?

Reply:
I just wanted to be on the safe side. In particular, my thinking here was that the `PostCreator` process is completed before updating the topic title.

Any drawback here if we enqueue after 5 seconds?